### PR TITLE
Use asserts to ensure that the predicate used in find function works.

### PR DESCRIPTION
**Pull request by ChenXy128**
…ons properly.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/PersonFieldsContainKeywordPredicate.java [hunk 1]
```diff
@@ -39,7 +39,9 @@ public PersonFieldsContainKeywordPredicate(Map<String, String> categoryDescripti
      */
     @Override
     public boolean test(Person person) {
+        assert categoryDescriptionMap != null;
```
Comment (NusMinato) [inline src/main/java/seedu/address/model/person/PersonFieldsContainKeywordPredicate.java:42]: Good job! This assertion can avoid the case that categoryDescriptionMap is null.

**Review by NusMinato — COMMENTED** [review]
Looks good to me. I like your assertion.

File: src/main/java/seedu/address/model/person/PersonFieldsContainKeywordPredicate.java [hunk 2]
```diff
@@ -39,7 +39,9 @@ public PersonFieldsContainKeywordPredicate(Map<String, String> categoryDescripti
...
         boolean matchesCategoryDescription = categoryDescriptionMap.entrySet().stream().anyMatch(entry -> {
+            assert entry.getKey() != null;
```
Comment (HongRay) [inline src/main/java/seedu/address/model/person/PersonFieldsContainKeywordPredicate.java:44]: Nice assert. I forgot to do it, am not sure whether they will still allow me to do my asserts.

File: src/main/java/seedu/address/model/person/PersonFieldsContainKeywordPredicate.java (hunk 1 above)
Comment (Xuyan0518) [inline src/main/java/seedu/address/model/person/PersonFieldsContainKeywordPredicate.java:42]: Nice job on adding the assertions to make sure the categoryDescriptionMap is not null.

**Review by Cedricaca — COMMENTED** [review]
nice la

File: docs/UserGuide.md [hunk 3]
```diff
@@ -93,8 +93,8 @@ Edits an existing person in the address book.
 Format: `edit INDEX [c/CATEGORY] [d/DESCRIPTION] [t/TAG]…​`
 
 * Edits the person at the specified `INDEX`. The index refers to the index number shown in the displayed person list. The index **must be a positive integer** 1, 2, 3, …​
-* The category you want to edit must **exist**. 
-* Description must be provided if you want to change a description for a category.  
+* The category you want to edit must **exist**.
```
Comment (Cedricaca) [inline docs/UserGuide.md:96]: i like this specification

**Review by Cedricaca — COMMENTED** [review]
very naise

**Comment by HongRay** [comment]
Too much conflicts!

**Pull request closed without merging**